slides.md

```diff
@@ -4,7 +4,148 @@
 
 ---
 # The tools
+
 ---
+# JavaScript basics
+
+To declare a variable, you can either use `const` (for constants), or `let` (for variables that can change over time).
+
+TLDR: Do not use `var`.
+
+Long explanation: `var` is a variable that can have multiple values as `let`, but it's visible in the outer scope (so: globally if outside a function, all the function inside a function). `let` will be visible only inside the scope it's declared in (e.g., a `for` loop inside a function).
+
+```javascript
+const count = 10;
+
+for (let i = 0; i < count; i++) {
+  console.log(`Hello ${count}`);
+}
+```
+
+You can use single, double, and inverse quotes to create string literals, only inverse quotes can interpolate variables though.
+
+```javascript
+const name = "Daniele";
+
+console.log("Hello " + name);
+console.log('Hello ' + name);
+console.log(`Hello ${name}`);
+```
+
+To create arrays, you can use square brackets `[]`.
+
+```javascript
+const anArray = [1, 2, 3];
+```
+
+To create objects (string indexed dictionaries), you can use curly brackets.
+```javascript
+const obj = {
+  a: 1,
+  b: "Hello"
+  c: {
+    nested: true
+  }
+};
+```
+
+To create a function, you can use both the `function` keyword, or use the arrow `=>` syntax. Arrow functions can directly return a value as an expression, and don't need `()` around a single argument. They need `()` around the return value if they return an object.
+
+TLDR: they're mostly interchangeable, but prefer arrow syntax over function.
+
+Long explanation: The main differences are:
+1. Arrow functions cannot be used as constructors (a function works as a class constructor of itself, kind of, `class` is just syntactical sugar).
+2. `this` in arrow functions is easier to reason about: it's the `this` of the scope in which the function is declared. In a normal function, `this` might change depending on how it's invoked.
+
+```javascript
+function say(name) {
+  console.log(`Hello ${name}`);
+};
+
+const say = (name) => {
+  console.log(`Hello ${name}`);
+};
+
+const double = n => n * 2;
+const makeConfig = () => ({debug: true, log: false}); // returns the object
+```
+
+You can _spread_ objects and arrays, which applies their content to the current expression. The order in which you spread your object **matters**.
+
+```javascript
+const numbers = [1, 2, 3];
+const add = (a, b, c) => a + b + c;
+const total = add(...numbers); // 6
+const twice = [...numbers, ...numbers]; // [1,2,3,1,2,3]
+
+const defaults = {
+  log: false,
+  debug: true
+}
+
+const options = {
+  debug: false
+}
+
+const overrideOptions = {...options, ...defaults} // {log: false, debug: true}
+const overrideDefaults = {...defaults, ...options} // {log: false, debug: false}
+const overrideWithFields = {...defaults, debug: false} // {log: false, debug: false}
+```
+
+You can **destructure** objects and arrays.
```

Comment:
Maybe a brief mention of what destructuring is? Or is it common knowledge for JS devs?

Reply:
818ab45